src/shared/navgen/nav.cpp

```diff
@@ -61,43 +83,52 @@ static NavgenStatus::Code CodeForFailedDtStatus(dtStatus status)
 
 static int tileSize = 64;
 
-void NavmeshGenerator::WriteFile() {
-	if ( d_->status.code == NavgenStatus::TRANSIENT_FAILURE )
+void NavmeshGenerator::WriteFile( const NavgenTask &t ) {
```

Comment:
Can we leave comments about which functions are main thread only here too?

Reply:
I added comments to functions which can be called from a non-main thread since there are much fewer of those and they are the ones with restrictions.